### ISSUE TITLE
plot(ctd) croaks on some NA condition

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1372]
### Short summary of problem

`plot,ctd-method` can croak on NA values.

### Details (optional)

This is from a worked exercise in OAR, so it ought to work :-)

### What you did

```R
library(oce)
data(ctdRaw)
bad <- function(x, n=2)
    is.na(x) | (abs(x-mean(x,na.rm=TRUE)) > n*sd(x,na.rm=TRUE))
for (i in 1:2) {
    b <- bad(ctdRaw[["salinity"]])|bad(ctdRaw[["temperature"]])
    ctdRaw[["flags"]]$salinity <- ifelse(b, 3, 2)
    ctdRaw[["flags"]]$temperature <- ifelse(b, 3, 2)
    ctdRaw <- handleFlags(ctdRaw)
}
plot(ctdRaw)
```

### What you expected to happen

A sweet plot.

### What happened

```R
Error in approx(l$p_mid, l$N2, p, rule = 2) : 
  approx(): attempted to interpolate NA values
Calls: plot -> plot -> .local -> plotProfile -> swN2 -> approx
Execution halted
```

### How urgent is this?

I'll fix this today or tomorrow morning.

### Output from sessionInfo()

```
R version 3.4.3 (2017-11-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: OS X El Capitan 10.11.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_CA.UTF-8/en_CA.UTF-8/en_CA.UTF-8/C/en_CA.UTF-8/en_CA.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] oce_0.9-23     gsw_1.0-5      testthat_2.0.0 landsat_1.0.8  rgdal_1.2-16   sp_1.2-5      

loaded via a namespace (and not attached):
 [1] compiler_3.4.3  magrittr_1.5    R6_2.2.2        Matrix_1.2-12   tools_3.4.3     mgcv_1.8-22     nlme_3.1-131    lmodel2_1.7-2  
 [9] grid_3.4.3      rlang_0.1.4     lattice_0.20-35
```

**Comment by dankelley** [comment]
Where it occurs--
```
> traceback()
6: approx(l$p_mid, l$N2, p, rule = 2) at sw.R#323
5: swN2(x, df = df, eos = eos) at ctd.R#5101
4: plotProfile(x, xtype = "density+N2", plim = plim, N2lim = N2lim, 
       densitylim = densitylim, eos = eos, useSmoothScatter = useSmoothScatter, 
       df = df, grid = grid, col.grid = "lightgray", lty.grid = "dotted", 
       cex = cex[w], pch = pch[w], type = type[w], keepNA = keepNA, 
       inset = inset, add = add, debug = debug - 1, ...) at ctd.R#3089
3: .local(x, ...)
2: plot(ctdRaw)
1: plot(ctdRaw)
```

**Comment by dankelley** [comment]
Fixed in "develop" branch commit aa155e9003f61e5f2a3518d807cea0ce29ea6901

PS. the N^2 looks terrible, but that's just how GSW rolls, baby.
